bin/uglifyjs

```diff
@@ -314,16 +314,43 @@ function run() {
     try {
         if (options.parse) {
             if (options.parse.acorn) {
+                var annotations = Object.create(null);
                 files = convert_ast(function(toplevel, name) {
-                    return require("acorn").parse(files[name], {
+                    var content = files[name];
+                    var list = annotations[name] = [];
+                    var prev = -1;
+                    return require("acorn").parse(content, {
                         allowHashBang: true,
                         ecmaVersion: "latest",
                         locations: true,
+                        onComment: function(block, text, start, end) {
+                            var match = /[@#]__PURE__/.exec(text);
+                            if (!match) {
+                                if (start != prev) return;
+                                match = [ list[prev] ];
```

Comment:
Should take care of https://github.com/mishoo/UglifyJS/pull/4879#issuecomment-828556915